src/libraries/Common/src/System/Security/Cryptography/ECCng.ImportExport.NamedCurve.cs

```diff
@@ -0,0 +1,197 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+using BCRYPT_ECCKEY_BLOB = Interop.BCrypt.BCRYPT_ECCKEY_BLOB;
+using ErrorCode = Interop.NCrypt.ErrorCode;
+using KeyBlobMagicNumber = Interop.BCrypt.KeyBlobMagicNumber;
+
+namespace System.Security.Cryptography
+{
+    internal static partial class ECCng
+    {
+        internal delegate TResult EncodeBlobFunc<TResult>(byte[] blob);
+
+        // When possible, operate on the blob inside the callback since the array will be pinned during its execution.
+        internal static T EncodeEccKeyBlob<T>(KeyBlobMagicNumber magic, byte[] x, byte[] y, byte[]? d, Func<byte[], T> encodeCallback, bool clearBlob = true)
+        {
+            bool includePrivateParameters = d is not null;
+            byte[] blob;
+            unsafe
+            {
+                // We need to build a key blob structured as follows:
+                //     BCRYPT_ECCKEY_BLOB   header
+                //     byte[cbKey]          Q.X
+                //     byte[cbKey]          Q.Y
+                //     -- Only if "includePrivateParameters" is true --
+                //     byte[cbKey]          D
+
+                int blobSize = sizeof(BCRYPT_ECCKEY_BLOB) +
+                    x.Length +
+                    y.Length;
```

Comment:
Somewhere there should be an assert that y.Length == x.Length, since there's only one `cbKey` value and it's assumed to match.

(I believe in the original location this was already done by ECPoint.Verify, or something like that)